### ISSUE TITLE
Hase eats away my CPU

### ISSUE THREAD

**Issue by Wuzzy2** [theZiz/hase#10]
I am using a gamer laptop on Parabola (Arch Linux derivate) yet the game it somewhat slow at times (low FPS). While the game runs, it makes one of the CPU cores go 100%.

Specs:
* Processor: Intel(R) Core(TM) i7-4510U CPU @ 2.00GHz
* Graphics: On chip
* RAM: 8 GiB

This just doesn't seem right, especially for a simple 2D game. I am able to run much more demanding games just fine.
Seems like Hase is pretty wasteful on CPU time.

It's not so bad that Hase is unplayable, but the fact that it eats my CPU feels wrong.

**Comment by theZiz** [comment]
Hase is rendered on GPU completely. Historic reasons. Nothing I will change in the future. However because of this the fps directly correlates with the chosen resoltuon. So just try a smaller window or directly use a smaller screen resolution to still being able seeing everything on a full screen.

Hase is running way better on small devices like the GCW Zero as  the ratio from full hd (1920 x 1080) to 320 x 240 is 1:27. So the PC needs 27 times as much computational CPU power. :open_mouth:

**Comment by klopsi** [comment]
I think Ziz meant to say hase is rendered on CPU completely.  I have profiled it and there is nothing in sparrow3d or hase I can easily make faster.

**Comment by theZiz** [comment]
Yes, you are right :)

**Comment by Wuzzy2** [comment]
Did you review the algorithmic complexity of the rendering routine?

**Comment by theZiz** [comment]
Yeah, it depends on the number of pixels. This is the reason why it works very well with very small screen devices such as the gp2x, is not much faster on way more powerful machines as the pandora and very slow on full HD screens. :wink:
